plugins/keycloak-plugin-token-provider/src/main/java/com/redhat/che/keycloak/token/provider/service/KeycloakTokenProvider.java

```diff
@@ -86,18 +82,20 @@ public String obtainGitHubToken(String keycloakToken)
    * @throws UnauthorizedException
    * @throws ServerException
    */
-  public String obtainOsoToken(String keycloakToken)
+  public String obtainOsoToken(final String keycloakToken)
       throws IOException, ServerException, UnauthorizedException, ForbiddenException,
           NotFoundException, ConflictException, BadRequestException {
-    String responseBody = getResponseBody(openShiftEndpoint, keycloakToken);
+    return getAccessToken(openShiftEndpoint, keycloakToken);
+  }
+
+  private String getAccessToken(final String endpoint, final String keycloakToken)
+      throws JsonProcessingException, IOException, ServerException, UnauthorizedException,
+          ForbiddenException, NotFoundException, ConflictException, BadRequestException {
+    String responseBody = getResponseBody(endpoint, keycloakToken);
     ObjectMapper mapper = new ObjectMapper();
     JsonNode json = mapper.readTree(responseBody);
-    JsonNode token = json.get(ACCESS_TOKEN);
-    if (token != null) {
-      return token.asText();
-    } else {
-      return null;
-    }
+    JsonNode accessToken = json.get(ACCESS_TOKEN);
+    return accessToken.asText();
```

Comment:
It can't have npe if accesstoken is not retrieved ?

Reply:
this should never happen I believe

Reply:
If you got 200 response with a JSON then correct. access_token should be always there. If there is no token available (user has not linked GitHub account) then there will be 401 response with a JSON error message and ""WWW-Authenticate" header with details what client should do to solve the problem (it should initiate account linking by calling our GitHub linking API).

Reply:
401 should fail in `getResponseBody` and exception will be thrown